frontend/src/app/app.component.ts

```diff
@@ -34,17 +36,22 @@ export class AppComponent {
             ) {
                 this.spinner.hide();
             }
+
             if (event instanceof RoutesRecognized) {
                 const route = event.state.root.firstChild;
 
                 if (route) {
-                    this.isRouteNotFound = route.routeConfig === null;
+                    const isFallbackRoute = route.routeConfig?.path === '**' && route.routeConfig?.pathMatch === 'full';
+
+                    this.isRouteFound = !isFallbackRoute;
                 }
             }
 
             if (event instanceof NavigationStart) {
-                this.isAuthPage = event.url.startsWith('/auth') || event.url === '/';
+                this.isWithoutHeaderPage = event.url.startsWith('/auth') || event.url === '/';
```

Comment:
I think it's not good to check header by path. 

I would recommend another approach: to add data to our routes. For example. 

```suggestion
    {
        path: 'main',
        loadChildren: () => import('./modules/main/main.module').then((m) => m.MainModule),
        canActivate: [AuthorizedGuard],
        data: {showHeader: true}
    },
```

and *ngIf=showHeader as well. 

Another option is to add HeaderGuard with data like in previous example and to add this guard to the pages where we need the header. 

You can choose one of it approaches or find even better, good luck.